apis/v1alpha2/validation/gateway.go

```diff
@@ -26,6 +27,19 @@ import (
 	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
+var (
+	exists          = struct{}{}
+	tcpUdpProtocols = map[gatewayv1a2.ProtocolType]struct{}{
```

Comment:
Is it not possible to use the set package from the k/k codebase

Reply:
Hey can you tell me the name of the package ?

Reply:
@bowei I think the package in k/k does have it for generic type and rather it is for primitive types